src/main/resources/org/jenkinsci/plugins/docker/workflow/Docker.groovy

```diff
@@ -80,21 +80,29 @@ class Docker implements Serializable {
             def parsedArgs = args.split(/ (?=([^"']*["'][^"']*["'])*[^"']*$)/)
             def dir = parsedArgs[-1] ?: '.'
 
-            // Detect custom Dockerfile:
+            // iterate through arguments to check for extra variables that need to be set
+            def target = null
             def dockerfile = "${dir}/Dockerfile"
             for (int i=0; i<parsedArgs.length; i++) {
                 def arg = parsedArgs[i]
-                if ((arg == '-f' || arg.startsWith('--file')) && i < (parsedArgs.length - 1)) {
+                // determine build target
+                if (arg.startsWith('--target=')) {
+                    target = arg.split('=')[1]
+                }
+                else if (arg.startsWith('--target') && i < (parsedArgs.length - 1)) {
+                    target = parsedArgs[i+1]
+                }
+                // determine custom dockerfile
+                else if ((arg == '-f' || arg.startsWith('--file')) && i < (parsedArgs.length - 1)) {
                     dockerfile = arg.startsWith('--file=') ? arg.split('=')[1] : parsedArgs[i+1]
```

Comment:
I'm pretty sure you should be `i++`ing somewhere in here.

Reply:
For most of the cases you're looking at a single element of the array and walking forward by one. For `-f` and `--file`, you're looking at the next element as well.

Consider: `['-f', '--target=Foo']`.
The correct parsing is to treat the second argument as a file. I think your code will treat it both as a file and as an argument indicating target of `Foo`.

Reply:
Ah, I see what you mean. I didn't write the `--file` parsing stuff so didn't bother to fix it even if I do think it's buggy, but will give it a second look.